packages/pipeline/src/scripts/pull_slippage.ts

```diff
@@ -0,0 +1,49 @@
+import { Connection, ConnectionOptions, createConnection } from 'typeorm';
+
+import { logUtils } from '@0x/utils';
+
+import { EdpsSource} from '../data_sources/slippage';
+import { Slippage } from '../entities';
+import * as ormConfig from '../ormconfig';
+import { calculateSlippage } from '../parsers/slippage';
+import { handleError } from '../utils';
+
+// Number of orders to save at once.
+const BATCH_SAVE_SIZE = 1000;
+
+// USD amounts for slippage depths
+// tslint:disable-next-line:custom-no-magic-numbers
+const USD_AMOUNTS = [10, 100, 1000, 10000];
+
+// TODO: fetch from database
+const TOKENS = ['BAT', 'DAI', 'FUN', 'MANA', 'OMG', 'REP', 'TUSD', 'ZRX', 'MKR', 'BNB', 'USDC'];
+
+let connection: Connection;
+
+(async () => {
+    connection = await createConnection(ormConfig as ConnectionOptions);
+    const edpsSource = new EdpsSource();
+
+    logUtils.log('Fetching slippage records');
+    const nestedSlippages: Slippage[][][] = await Promise.all(TOKENS.map(async symbol => {
+        const usdPrice = await edpsSource.getUsdPriceAsync(symbol);
+        return Promise.all(USD_AMOUNTS.map(async usdAmount => {
+            const amount = usdAmount / usdPrice;
+            const buyEdps = await edpsSource.getEdpsAsync('buy', symbol, amount);
+            const sellEdps = await edpsSource.getEdpsAsync('sell', symbol, amount);
+            return Array.from(buyEdps.keys()).map(exchange => {
+                const slippage: Slippage = calculateSlippage(usdAmount, exchange, buyEdps, sellEdps);
+                return slippage;
+            });
+        }));
+    }));
+    const slippagesWithEmptyRecords = nestedSlippages
```

Comment:
What is this doing? is this a `flatten`? https://ramdajs.com/docs/#flatten

Reply:
Yes, indeed! This is just my awkward Typescript/Javascript - will change to `flatten`

Reply:
Quick note on this - I tried using Ramda but it didn't really flatten as expected. Maybe because the nested arrays contained objects (?). I've left it as is for now.

Reply:
Does unnest work?